src/lib/hooks/useStore.js

```diff
@@ -6,7 +6,14 @@ const useStore = create(
 	persist(
 		set => ({
 			myVehicles: [],
-			vinFound: '',
+			vinValidity: false,
+			toggleVinValidity: () => {
+				set(state => ({ vinValidity: !state.vinValidity }));
+			},
+			displayVinError: false,
+			toggleDisplayVinError: () => {
+				set(state => ({ displayVinError: !state.displayVinError }));
+			},
```

Comment:
This part focusses on validating your VINS. Chances are that there is a way to do that without having two separate states von vinValidity and displayVinError. Isn't it reason enough to render the error if vinValidity is false?

Reply:
Yep, absolutely right.. I cleaned it up and solved it via set states :)